src/platform/tls_mitls.c

```diff
@@ -1980,6 +2014,39 @@ QuicTlsOnTicketReady(
     }
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+QuicTlsSendTicket(
+    _In_ QUIC_TLS* TlsContext,
+    _In_ uint32_t EncodedTicketLength,
+    _In_reads_bytes_(EncodedTicketLength)
+        const uint8_t* EncodedTicket
+    )
+{
+    QUIC_DBG_ASSERT(TlsContext->IsServer);
+    QUIC_DBG_ASSERT((EncodedTicketLength > 0 && EncodedTicket != NULL) ||
+        (EncodedTicketLength == 0 && EncodedTicket == NULL));
+
+    QuicTraceLogConnVerbose(
+        miTlsSend0RttTicket,
+        TlsContext->Connection,
+        "Sending 0-RTT ticket");
+    if (!FFI_mitls_quic_send_ticket(TlsContext->miTlsState, EncodedTicket, EncodedTicketLength)) {
+        QuicTraceEvent(
+            TlsError,
+            "[ tls][%p] ERROR, %s.",
+            TlsContext->Connection,
+            "FFI_mitls_quic_send_ticket failed");
+        return QUIC_STATUS_TLS_ERROR;
+    }
+
+    //
+    // Indicate data is ready from TLS.
+    //
+    TlsContext->ProcessCompleteCallback(TlsContext->Connection);
```

Comment:
Your design assumes that async process is supported by TLS. miTLS is the only one that actually does support this.